### PR TITLE
Fix 8427

### LINKED ISSUE #8427: Old way to specify resources in kubernetes_pod_operator.py
## Is your feature request related to a problem? Please describe.
Yes.

Perhaps the method shown on the page below is outdated.
```Python
resources={'limit_memory': "250M", 'limit_cpu': "100m"},
```
https://cloud.google.com/composer/docs/composer-2/use-kubernetes-pod-operator#kubernetespodoperator_configuration

In the `composer-2.0.28-airflow-2.3.3` this method receives an error.
```
Traceback (most recent call last):
  File "/opt/python3.8/lib/python3.8/site-packages/airflow/models/baseoperator.py", line 858, in __init__
    self.resources = coerce_resources(resources)
  File "/opt/python3.8/lib/python3.8/site-packages/airflow/models/baseoperator.py", line 133, in coerce_resources
    return Resources(**resources)
TypeError: __init__() got an unexpected keyword argument 'limit_memory'
```

## Describe the solution you'd like.
Update the documentation

new way
```Python
container_resources = k8s.V1ResourceRequirements(
      limits={'memory': '250M', 'cpu': "100m"},
)
```

## Additional context.
 I referred to the following history
https://github.com/apache/airflow/blob/ea55626d79fdbd96b6d5f371883ac1df2a6313ec/kubernetes_tests/test_kubernetes_pod_operator.py#L366

Thanks!

### PR DESCRIPTION
**Pull request by leahecole**
## Description

Fixes #8427
In this, we only care about a successful "ex-all-configs" task run. The other three failed tasks shown are expected and are because of things I didn't do in that Composer environment, not because of DAG issues.
image
Here's an example of a successful run of the gke_operator
image

I had to specify the "perl" image to match the kubernetes_pod_operator example - if we don't have it, a weird weird error occurs and the pod fails to start. 

Normally I don't bump constraints, but I did bump this one to match with what we offer in Composer https://cloud.google.com/composer/docs/concepts/versioning/composer-versions

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

File: composer/workflows/kubernetes_pod_operator.py [hunk 1]
```diff
@@ -19,8 +19,10 @@
 
 from airflow import models
 from airflow.kubernetes.secret import Secret
-from airflow.providers.cncf.kubernetes.operators.kubernetes_pod import KubernetesPodOperator
-
+from airflow.providers.cncf.kubernetes.operators.kubernetes_pod import (
+    KubernetesPodOperator,
+)
+from kubernetes.client import models as k8s
```
Comment (dandhlee) [inline composer/workflows/kubernetes_pod_operator.py:25]: if `kubernetes.client.models` is synonymous to `k8s` then totally fine to leave as is, but wanted to make sure it's fully descriptive.
```suggestion
from kubernetes.client import models as k8_models
```

**Review by dandhlee — APPROVED** [review]
LGTM, one minor nit.

File: composer/workflows/kubernetes_pod_operator.py [hunk 2]
```diff
@@ -85,151 +89,160 @@
...
-        # resources = pod.Resources() instead passing a dict
-        # For more info see:
-        # https://github.com/apache/airflow/pull/4551
-        resources={'limit_memory': "250M", 'limit_cpu': "100m"},
+        # Prior to Airflow 2.3 and the cncf providers package 5.0.0
+        # resources were passed as a dictionary. This change was made in
+        # https://github.com/apache/airflow/pull/27197
+        # Additionally, "memory" and "cpu" were previously named
+        # "limit_memory" and "limit_cpu"
+        #resources={'limit_memory': "250M", 'limit_cpu': "100m"},
```
Comment (dandhlee) [inline composer/workflows/kubernetes_pod_operator.py:239]: Lint complaints :P
```suggestion
        # resources={'limit_memory': "250M", 'limit_cpu': "100m"},
```

**Review by dandhlee — COMMENTED** [review]
Running into an additional failure:
```
airflow.exceptions.AirflowException: Invalid arguments were passed to KubernetesPodOperator (task_id: ex-all-configs). Invalid arguments were:
**kwargs: {'container_resources': {'limits': {'cpu': '100m', 'memory': '250M'}, 'requests': None}}
```

**Comment by leahecole** [comment]
I will look into why the test fails when it passes in my airflow environment!

File: composer/workflows/kubernetes_pod_operator.py (hunk 1 above)
Comment (leahecole) [reply composer/workflows/kubernetes_pod_operator.py]: sure, made that change. i had it matching what I saw in the test, but more descriptive names never hurt 😄

**Comment by leahecole** [comment]
ok great! that issue is fixed. Once Rach is back in office I'll have her look it over :)

**Review by rachael-ds — APPROVED** [review]
LGTM

**Pull request merged**